### PR TITLE
Namespace component name

### PR DESCRIPTION
**Pull request by florentferry**
#12

### CODE REVIEW

**Comment by Spone** [comment]
@florentferry I think we should add an `index.js` file in each namespace, and import it from `frontend/components/index.js`

For instance, for a component name `admin/header`, we would have:

```js
// frontend/components/index.js

import "components/admin";
```
```js
// frontend/components/admin/index.js

import "components/admin/header/header";
```

**Comment by florentferry** [comment]
@Spone Running`rails generate component admin/sub/button` generates a correct namespacing. That's resolution seems good to you ?

```js
// frontend/components/index.js

import "components/admin";
```

```js
// frontend/components/admin/index.js

import "components/admin/sub";
```

```js
// frontend/component/admin/sub/index.js

import "components/admin/sub/button/button";
```

**Comment by florentferry** [comment]
@Spone If we follow this path, then `button` should have `index.js` instead of `button.js`, no ?

```js
// frontend/components/admin/sub/index.js

import "components/admin/sub/button";
```

**Comment by Spone** [comment]
Yeah but in this case we make an exception in order to have more explicit file names (else you would have only `index.js` files open in your editor and it's pretty confusing).

We only use `index.js` as an index of all JS files to import.

File: lib/generators/component/component_generator.rb [hunk 1]
```diff
@@ -28,10 +28,34 @@ def create_locale_files
     end
   end
 
-  def append_frontend_packs
-    append_to_file "frontend/components/index.js" do
-      "import \"components/#{split_name.join("/")}/#{component_name}\";\n"
+  def import_to_packs
+    root_path = Pathname.new("frontend")
+    base_path = root_path + "components"
```
Comment (Spone) [inline lib/generators/component/component_generator.rb]: Isn't it better to do `File.join(root_path, "components")`?

Comment (florentferry) [reply lib/generators/component/component_generator.rb]: https://ruby-doc.org/stdlib-1.9.3/libdoc/pathname/rdoc/Pathname.html#method-i-2B

**Pull request merged**